tests/test-derivation.nix

```diff
@@ -79,20 +78,60 @@ let
     # At least one of these should be undefined
     assert lib.assertMsg (tests == [ ] || module == null) "Both module & tests can't be supplied";
     let
+      helpers = import ../lib/helpers.nix {
+        inherit pkgs lib;
+        _nixvimTests = true;
+      };
       moduleAsList = lib.optional (module != null) { inherit module name; };
       moduleList = tests ++ moduleAsList;
       testDerivations = builtins.map (
         {
           module,
           name,
           dontRun ? false,
-        }:
-        {
-          derivation = makeNixvimWithModule {
-            inherit pkgs module extraSpecialArgs;
-            _nixvimTests = true;
+        }@args:
+        let
+          result = lib.evalModules {
+            modules = [
+              module
+              ./_module.nix
+              ../modules/top-level
+            ] ++ lib.optional (args ? dontRun) { tests.dontRun = lib.mkDefault dontRun; };
+            specialArgs = helpers.modules.specialArgsWith extraSpecialArgs;
           };
-          inherit name dontRun;
+
+          # TODO: allow "expecting" specific errors
+          inherit (result.config) warnings;
+          assertions = lib.pipe result.config.assertions [
+            (lib.filter (x: !x.assertion))
+            (lib.map (x: x.message))
+          ];
+
+          errors = pkgs.runCommand name { inherit name assertions warnings; } ''
+            echo "Issues found evaluating $name":
+            if [ -n "$assertions" ]; then
+              echo "Unexpected assertions:"
+              for it in "$assertions"; do
+                echo "- $it"
+              done
+              echo
+            fi
+            if [ -n "$warnings" ]; then
+              echo "Unexpected warnings:"
+              for it in "$warnings"; do
+                echo "- $it"
+              done
+              echo
+            fi
+            exit 1
+          '';
```

Comment:
Should this be done on a per-module basis or on a per moduleList basis?

If the latter, how should we handle some modules having errors and others needing to be build/run?